File: python-impl/insfrastructure/usecases/deposit_usecase.py
from insfrastructure.repositories.bank_account_repository import BankAccountRepository


class DepositToAccountUseCase:
    def __init__(self, bank_account_repo: BankAccountRepository):
        self.bank_account_repo = bank_account_repo

    def execute(self, account_id: int, amount: float) -> bool:
        return self.bank_account_repo.deposit(account_id, amount)
        

File: typescript-impl/src/infrastructure/usecases/deposit_usecase.ts
import { BankAccountRepository } from '../repositories/bank_account_repository';

export class DepositToAccountUseCase {
    constructor(private bank_account_repo: BankAccountRepository) { }

    execute(account_id: number, amount: number): boolean {
        return this.bank_account_repo.deposit(account_id, amount);
    }
}
